### DISCUSSION TITLE
Question: authenticate on batch mode without user interaction

### DISCUSSION THREAD

**Discussion by fmertins** [forcedotcom/dataloader#737] [Q&A] [answered]
Hello, we're using DL being called from PowerShell scripts (that is, DL "batch mode" with "process conf" XML file definitions) and whenever it processes a "bean" it shows up the browser asking to authenticate.

Is there a way to configure only once a Connected App together DL so everytime DL runs it would already know the necessary credentials? Hence, allowing a better automated integration without user interaction?

Thank you in advance!

**Comment by ashitsalesforce** [comment]
Hi @fmertins , have you specified sfdc.username and sfdc.password properties in process-conf.xml as documented [here](https://developer.salesforce.com/docs/atlas.en-us.dataLoader.meta/dataLoader/loader_params.htm)?

**Comment by fmertins** [comment]
Hi, yes... But the thing is, the Org is configured with SSO Login, so unfortunately we can't use regular credentials username+password I'm afraid.

**Comment by ashitsalesforce** [comment, marked as answer]
Hi @fmertins ,

A customer had a similar issue. They [wrote a wrapper script](https://github.com/forcedotcom/dataloader/issues/385#issuecomment-1000130579) that gets the OAuth access token from the IdP and passes the access token to data loader as a command line option `sfdc.oauth.accesstoken`.  Will a similar approach work for you?

**Comment by fmertins** [comment]
We'll take a look at that approach, thanks a lot!!
